src/plugins/intel_cpu/src/infer_request.cpp

```diff
@@ -257,6 +264,38 @@ void InferRequestBase::changeDefaultPtr() {
         auto output = outputNodesMap.find(it.first);
         if (output != outputNodesMap.end()) {
             auto parentEdge = output->second->getParentEdgeAt(0);
+            if (Graph::Status::ReadyDynamic == graph->getStatus()) {
+                bool canBeInPlace = true;
+                // TODO: filter
+
+                // share intel_cpu::Tensor to Graph by injecting to corresponding ProxyMemoryMngr instance.
+                ProxyMemoryMngrPtr outputMemMngr;
+                const auto &outMemMngrMap = graph->outputNodesMemMngrMap;
+                auto itr = outMemMngrMap.find(it.first);
+                if (itr != outMemMngrMap.end()) {
+                    outputMemMngr = itr->second;
+                    OPENVINO_ASSERT(outputMemMngr, "proxy mem manager for output ", it.first, " is empty.");
+                } else {
+                    canBeInPlace = false;
+                    DEBUG_LOG("no proxy mem manager for output ", it.first, " !");
+                }
+
+                if (canBeInPlace) {
+                    auto tt = std::get<0>(outputsTensor2BlobMap[it.first]);  // there is no way to get tensor from blob.
+                    auto memptr = tt->get_memory();
+                    outputMemMngr->reset(memptr->getMemoryMngr());
+                    DEBUG_LOG("reset proxy ", outputMemMngr, ", actual ", memptr->getMemoryMngr(), " graph ", graph, " inferrequest ", this);
+                    DEBUG_LOG(it.first, ", blob ", std::get<1>(outputsTensor2BlobMap[it.first]), ", tensor ", tt);
+                } else {
+                    if (outputMemMngr) {
+                        outputMemMngr->reset(nullptr);
+                        DEBUG_LOG("reset nullptr", " graph ", graph, " inferrequest ", this);
+                    }
+                }
+
+                continue;
```

Comment:
```suggestion
                // share intel_cpu::Tensor to Graph by injecting to corresponding ProxyMemoryMngr instance.
                const auto &outMemMngrMap = graph->outputNodesMemMngrMap;
                auto itr = outMemMngrMap.find(it.first);
                if (itr != outMemMngrMap.end()) {
                    auto outputMemMngr = itr->second;
                    OPENVINO_ASSERT(outputMemMngr, "proxy mem manager for output ", it.first, " is empty.");
                    if (canBeInPlace) {
                        auto tt = std::get<0>(outputsTensor2BlobMap[it.first]);  // there is no way to get tensor from blob.
                        auto memptr = tt->get_memory();
                        outputMemMngr->reset(memptr->getMemoryMngr());
                        DEBUG_LOG("reset proxy ", outputMemMngr, ", actual ", memptr->getMemoryMngr(), " graph ", graph, " inferrequest ", this);
                        DEBUG_LOG(it.first, ", blob ", std::get<1>(outputsTensor2BlobMap[it.first]), ", tensor ", tt);
                     } else {
                        outputMemMngr->reset(nullptr);
                        DEBUG_LOG("reset nullptr", " graph ", graph, " inferrequest ", this);
                     }
                 }

                continue;
```
We shouldn't process the output memory if it's mem manager absents in the `outMemMngrMap`, should we?